File: 11-Title-Case-Sentence.js
// using map + uppercase
function titleCase(str) {
    return str.toLowerCase().split(' ')
              .map(word => word[0].toUpperCase() + word.slice(1))
              .join(' ')
}

// using regexp
// says "match each first letter after (beginning of string or whitespace"
function titleCase(str) {
    return str.toLowerCase().replace(/(^|\s)(\w)/g, (letter) => letter.toUpperCase())
}

console.log(titleCase("I'm a little tea pot"))

File: 11-Title-Case-Sentence.py
def titleCase(string):
    list_of_words = string.split()
    list_of_capitalized_words = [word[0].upper() + word[1:] for word in list_of_words]
    return ' '.join(list_of_capitalized_words)


# single line version
def titleCase(string):
    return ' '.join([(word[0].upper() + word[1:]) for word in string.split()])


# === tests ===
print(titleCase("I'm a little tea pot"))